[bug] start pouchd panic

### Ⅰ. Issue Description
Use latest pouch master code build, and start failed:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x90 pc=0x12867d6]

goroutine 124 [running]:
github.com/alibaba/pouch/daemon/containerio.(*ContainerIO).Read(0x0, 0xc420940000, 0x8000, 0x8000, 0x0, 0x26501922400, 0x1)
	/tmp/pouchbuild/src/github.com/alibaba/pouch/daemon/containerio/container_io.go:194 +0x26
io.copyBuffer(0x7f425f669358, 0xc42005bb60, 0x24c4400, 0x0, 0xc420940000, 0x8000, 0x8000, 0x19b38c0, 0xc4202aef00, 0x7f425f669358)
	/usr/local/go/src/io/io.go:392 +0x123
io.CopyBuffer(0x7f425f669358, 0xc42005bb60, 0x24c4400, 0x0, 0xc420940000, 0x8000, 0x8000, 0x8, 0x1b4c75a, 0x0)
	/usr/local/go/src/io/io.go:373 +0x82
github.com/alibaba/pouch/vendor/github.com/containerd/containerd/cio.copyIO.func2(0xc42091ebd0, 0x7f425f6692f8, 0xc42005bb60)
	/tmp/pouchbuild/src/github.com/alibaba/pouch/vendor/github.com/containerd/containerd/cio/io_unix.go:58 +0xf7
created by github.com/alibaba/pouch/vendor/github.com/containerd/containerd/cio.copyIO
	/tmp/pouchbuild/src/github.com/alibaba/pouch/vendor/github.com/containerd/containerd/cio/io_unix.go:55 +0x26b
```

### Ⅱ. Describe what happened


### Ⅲ. Describe what you expected to happen


### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1.
2.
3.

### Ⅴ. Anything else we need to know?


### Ⅵ. Environment:

- pouch version (use `pouch version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

--- a/cli/run.go
+++ b/cli/run.go
@@ -68,6 +68,7 @@ func (rc *RunCommand) runRun(args []string) error {
 		config.Cmd = args[1:]
 	}
 	containerName := rc.name
+	config.ContainerConfig.OpenStdin = rc.stdin
 
 	ctx := context.Background()
 	apiClient := rc.cli.Client()

--- a/ctrd/container.go
+++ b/ctrd/container.go
@@ -12,7 +12,6 @@ import (
 	"github.com/alibaba/pouch/pkg/errtypes"
 
 	"github.com/containerd/containerd"
-	"github.com/containerd/containerd/cio"
 	"github.com/containerd/containerd/errdefs"
 	"github.com/containerd/containerd/leases"
 	"github.com/containerd/containerd/linux/runctypes"
@@ -164,7 +163,7 @@ func (c *Client) RecoverContainer(ctx context.Context, id string, io *containeri
 		return errors.Wrap(err, "failed to load container")
 	}
 
-	task, err := lc.Task(ctx, cio.WithAttach(io.Stdin, io.Stdout, io.Stderr))
+	task, err := lc.Task(ctx, containerio.WithAttach(io.Stdin, io.Stdout, io.Stderr))
 	if err != nil {
 		if !errdefs.IsNotFound(err) {
 			return errors.Wrap(err, "failed to get task")

--- a/daemon/containerio/cio.go
+++ b/daemon/containerio/cio.go
@@ -2,6 +2,7 @@ package containerio
 
 import (
 	"context"
+	"fmt"
 	"io"
 	"io/ioutil"
 	"os"
@@ -45,18 +46,8 @@ func (c *cio) Close() error {
 	return c.closer.Close()
 }
 
-// FIFOSet is a set of fifos for use with tasks
-type FIFOSet struct {
-	// Dir is the directory holding the task fifos
-	Dir string
-	// In, Out, and Err fifo paths
-	In, Out, Err string
-	// Terminal returns true if a terminal is being used for the task
-	Terminal bool
-}
-
 // NewFifos returns a new set of fifos for the task
-func NewFifos(id string, stdin bool) (*FIFOSet, error) {
+func NewFifos(id string, stdin bool) (*containerdio.FIFOSet, error) {
 	root := "/run/containerd/fifo"
 	if err := os.MkdirAll(root, 0700); err != nil {
 		return nil, err
@@ -65,7 +56,7 @@ func NewFifos(id string, stdin bool) (*FIFOSet, error) {
 	if err != nil {
 		return nil, err
 	}
-	fifos := &FIFOSet{
+	fifos := &containerdio.FIFOSet{
 		Dir: dir,
 		In:  filepath.Join(dir, id+"-stdin"),
 		Out: filepath.Join(dir, id+"-stdout"),
@@ -109,7 +100,7 @@ func (g *wgCloser) Cancel() {
 	g.cancel()
 }
 
-func copyIO(fifos *FIFOSet, ioset *ioSet, tty bool) (_ *wgCloser, err error) {
+func copyIO(fifos *containerdio.FIFOSet, ioset *ioSet, tty bool) (_ *wgCloser, err error) {
 	var (
 		f           io.ReadWriteCloser
 		set         []io.Closer
@@ -182,6 +173,33 @@ func NewIOWithTerminal(stdin io.Reader, stdout, stderr io.Writer, terminal bool,
 		}()
 		cfg := containerdio.Config{
 			Terminal: terminal,
+			Stdout:   paths.Out,
+			Stderr:   paths.Err,
+			Stdin:    paths.In,
+		}
+		i := &cio{config: cfg}
+		set := &ioSet{
+			in:  stdin,
+			out: stdout,
+			err: stderr,
+		}
+		closer, err := copyIO(paths, set, cfg.Terminal)
+		if err != nil {
+			return nil, err
+		}
+		i.closer = closer
+		return i, nil
+	}
+}
+
+// WithAttach attaches the existing io for a task to the provided io.Reader/Writers
+func WithAttach(stdin io.Reader, stdout, stderr io.Writer) containerdio.Attach {
+	return func(paths *containerdio.FIFOSet) (containerdio.IO, error) {
+		if paths == nil {
+			return nil, fmt.Errorf("cannot attach to existing fifos")
+		}
+		cfg := containerdio.Config{
+			Terminal: paths.Terminal,
 			Stdout:   paths.Out,
 			Stderr:   paths.Err,
 			Stdin:    paths.In,

